daffodil-japi/src/main/scala/org/apache/daffodil/japi/io/InputSourceDataInputStream.scala

```diff
@@ -43,5 +43,13 @@ class InputSourceDataInputStream private[japi] (private [japi] val dis: SInputSo
   /**
    * Create an InputSourceDataInputStream from a byte array
    */
-  def this(arr: Array[Byte]) = this(SInputSourceDataInputStream(arr)) 
+  def this(arr: Array[Byte]) = this(SInputSourceDataInputStream(arr))
+
+  /**
+   * Used to see if there is more data or not after a parse operation.
+   * Whether more data is a left-over-data error, or just more data to
+   * parse in a subsequent parse call depends on the intention of the
+   * code calling the parse operation.
+   */
```

Comment:
I think it's worth adding a comment saying that this will block until either at least either n bytes are read or EOD is hit.

Reply:
I will update the comment.